ringhold/src/main/java/org/opencadc/ringhold/InventoryValidator.java

```diff
@@ -80,25 +86,71 @@
  */
 public class InventoryValidator implements Runnable {
     private static final Logger log = Logger.getLogger(InventoryValidator.class);
-    
-    private static final String FILTER_FILENAME = "artifact-deselector.sql";
 
-    private final ArtifactDAO iteratorDAO;
+    private final ArtifactDAO artifactIteratorDAO;
     private final ArtifactDAO artifactDAO;
+    private final String deselector;
     
     public InventoryValidator(Map<String, Object> txnConfig, Map<String, Object> iterConfig) { 
         this.artifactDAO = new ArtifactDAO();
         artifactDAO.setConfig(txnConfig);
         
-        this.iteratorDAO = new ArtifactDAO();
-        iteratorDAO.setConfig(iterConfig);
-        
-        // TODO: read FILTER_FILENAME, check and fail here
-        
+        this.artifactIteratorDAO = new ArtifactDAO();
+        artifactIteratorDAO.setConfig(iterConfig);
+
+        ArtifactDeselector artifactDeselector = new ArtifactDeselector();
+        try {
+            this.deselector = artifactDeselector.getConstraint();
+        } catch (ResourceNotFoundException ex) {
+            throw new IllegalArgumentException("missing required configuration: "
+                                                   + ArtifactDeselector.SQL_FILTER_FILE_NAME, ex);
+        } catch (IOException ex) {
+            throw new IllegalArgumentException("unable to read config: " + ArtifactDeselector.SQL_FILTER_FILE_NAME, ex);
+        }
     }
 
+    /**
+     * Find an artifact with a uri pattern in the deselector,
+     * delete the artifact and generate a deleted storage location event.
+     */
     @Override
     public void run() {
-        throw new UnsupportedOperationException("NOT IMPLEMENTED");
+        final TransactionManager transactionManager = this.artifactDAO.getTransactionManager();
+        final DeletedStorageLocationEventDAO deletedStorageLocationEventDAO =
+            new DeletedStorageLocationEventDAO(this.artifactDAO);
+
+        ResourceIterator<Artifact> artifactIterator = this.artifactIteratorDAO.iterator(this.deselector, null);
```

Comment:
Used the try-with-resources and added the catch for the IOException to be more explicit where and why it's thrown, rather than declare run() throws an IOException and let the Main() try/catch would handle it.

Reply:
I think the only source of IOException is in the close() call: you don't want to throw the RuntimeException on line 157. If the close() fails, it is because something else went wrong and was probably caught up at line 139 or maybe from line 151... you want that exception to get thrown. A throw-overrides-throw is a common way to hide errors - this is the one scenario where you log an exception only.

BTW, why doesn't the throw on line 145 require run() to declare that it throws?

Reply:
Updated to only catch and log the IOException for try-with-resources, removed the throws for a RuntimeException.
// BTW, why doesn't the throw on line 145 require run() to declare that it throws?
Good question, had to look that up. Short answer is no checked exceptions are thrown in the try that could be thrown again in the catch.